org.endless.dragonapprentice.xml

```diff
@@ -0,0 +1,5 @@
+"app-id": "org.endless.dragonapprentice",
+"runtime": "",
+"runtime-version": "1.0",
+"sdk": "",
+"command": "endless-dragonapprentice",
```

Comment:
Your manifest is almost empty. There is no module at all. Moreover, the filename should be `org.endless.dragonapprentice.yaml` or `org.endless.dragonapprentice.yml`, not `org.endless.dragonapprentice.xml`.

Reply:
Sorry. Where can I find information about the manifest format? I read this:
https://github.com/flathub/flathub/wiki/App-Requirements

...but I'm not sure what modules I need to include. This is a video game made with the Unity engine.

Reply:
> Sorry. Where can I find information about the manifest format? I read this:
> https://github.com/flathub/flathub/wiki/App-Requirements
> 
> ...but I'm not sure what modules I need to include. This is a video game made with the Unity engine.

URL? Is there a project website? You need at least a public available archive or Git/Svn/Bzr repo.

Anyway, why do you want to create a flatpak package? I mean, Unity games are self-contained and almost distro-agnostic. There is no need to create any kind of package and users usually don't like this. Just provide a TAR archive (e.g. `*.tar.gz`, `*.tar.bz2` or `*.tar.xz`) and you will be fine.
Personally, I think that [Game Jolt](https://gamejolt.com) and [Itch.io](https://itch.io) are better places for a such project.

Moreover, do you (or the vendor developer) own the `endless.org` domain? You should avoid a domain for which you don't have rights, unless it points to the code hosting service like GitHub, GitLab or SourceForge.

Reply:
Uploading it here is a petition from the vendor. Like you said, it was not org.endless but com.endlessm. The vendor own that domain.

I just received this xml:
https://justpaste.it/244tg

Which I convert to this yaml:
https://justpaste.it/6ptgf

I also receive this .json:
https://justpaste.it/69yr3

That manifest could be enough for a new pull request? Our repo is private, I don't know if that's a issue. Sorry for the inconvenience.

Reply:
> Uploading it here is a petition from the vendor.

Why the vendor did not submit it by yourself?

> Like you said, it was not org.endless but com.endlessm. The vendor own that domain.

Then the main file should be `com.endlessm.DragonsApprentice.json`, `com.endlessm.DragonsApprentice.yaml` or `com.endlessm.DragonsApprentice.yml`.

> I just received this xml:
> https://justpaste.it/244tg

The is the AppData file. It should stay XML. The filename for this should be `com.endlessm.DragonsApprentice.appdata.xml`.

> Which I convert to this yaml:
> https://justpaste.it/6ptgf

The AppData file must remain in XML format.

> I also receive this .json:
> https://justpaste.it/69yr3

This is the actual manifest. The filename for this should be `com.endlessm.DragonsApprentice.json`. You can convert it to YAML, but there is no need for this.

> That manifest could be enough for a new pull request? Our repo is private, I don't know if that's a issue. Sorry for the inconvenience.

Unfortunately, no. The manifest refers to several files that have not been attached:
 * `DragonsApprentice.zip` - the game archive
 * `com.endlessm.DragonsApprentice.desktop` - the desktop file
 * `com.endlessm.DragonsApprentice.png` - the icon

How can Flathub build a package if it has no access to these files?

Moreover, Flathub is about "stable releases, reproducible builds".
https://github.com/flathub/flathub/wiki/App-Requirements#stable-releases-reproducible-builds
Instead of attaching the game archive (`DragonsApprentice.zip`) to this repo, you should make it available online. Ideally, the URL should refer to the game version (e.g. `https://endlessm.com/DragonsApprentice/1.0/DragonsApprentice.zip` or `https://github.com/endlessm/DragonsApprentice/archive/Release_1.0.tar.gz`).

Please put the missing files (`com.endlessm.DragonsApprentice.json`, `com.endlessm.DragonsApprentice.appdata.xml`, `com.endlessm.DragonsApprentice.desktop`, `com.endlessm.DragonsApprentice.png`) to this repo, remove the invalid manifest (`org.endless.dragonapprentice.xml`), and change `"path": "DragonsApprentice.zip"` to the URL I mentioned earlier. Then you will get a proper review.

Reply:
Thanks, I do have those files, I only uploaded the manifest because that was the only file specified in the App Submission page (https://github.com/flathub/flathub/wiki/App-Submission) but I could upload everything else, including the game archive. I know the URL is prefered, but is still an option to attach the game archive to this repo, correct? We would prefer that if there's no problem.

Reply:
> Thanks, I do have those files, I only uploaded the manifest because that was the only file specified in the App Submission page (https://github.com/flathub/flathub/wiki/App-Submission) but I could upload everything else, including the game archive.

Well, now you know that you have to include everything that is required to build a flatpak package. :)

> I know the URL is prefered, but is still an option to attach the game archive to this repo, correct? We would prefer that if there's no problem.

Technically it shouldn't be a problem, but I am afraid that the Flathub maintainers will not like this.

@alexlarsson @nedrichards 
Would you like to comment this?

Anyway, you can easily create a new repo and upload the game archive (`DragonsApprentice.zip`) there. Of course, it could be a binary blob. Providing sources is not necessary. You can also put there the desktop file (`com.endlessm.DragonsApprentice.desktop`) and icon (`com.endlessm.DragonsApprentice.png`), as well as AppData file (`org.endless.dragonapprentice.xml`). The license file will also be welcomed. This will allow others to create Snap/RPM/DEB/PKGBUILD/ebuild/etc. packages and/or installers.

Reply:
Alright I think it would be better to create a new repo and upload the game archive, .desktop, icon and appdata there. Do I need to close this pull request and create a new one? Thanks.

Reply:
> Alright I think it would be better to create a new repo and upload the game archive, .desktop, icon and appdata there. Do I need to close this pull request and create a new one? Thanks.

There is no need to close this PR. Just put the correct manifest here (`com.endlessm.DragonsApprentice.json` - based on this: https://justpaste.it/69yr3) and remove the current one (`org.endless.dragonapprentice.xml`).

Reply:
I was unable to upload the game archive to a github repository due to file size limitation. A dropbox link like this would work?: https://www.dropbox.com/s/id2zldbkidrhynm/DragonsApprentice.zip

Reply:
I just changed the manifest file and added the sources url's. Thanks for your patience.